src/NodeAutoCompleteViewExtension/ViewModels/NodeAutoCompleteBarViewModel.cs

```diff
@@ -872,7 +887,25 @@ internal void AddCluster(ClusterResultItem ClusterResultItem)
 
             // AutoLayout should be called after all nodes are connected.
             NodeAutoCompleteUtilities.PostAutoLayoutNodes(wsViewModel.DynamoViewModel.CurrentSpace, node.NodeModel, clusterNodesModel.Select(x => x.NodeModel), false, false, false, finalizer);
+
+            //record all node and wire creation as one undo
+            RecordUndoModels(wsViewModel.Model, newNodesAndWires);
+        }
+
+        private void RecordUndoModels(WorkspaceModel workspace, List<ModelBase> undoItems)
```

Comment:
Saw a similar implementation somewhere else, maybe we dont have to repeat it and can put this as a utility function in the DynamoUtility class/project?